spring-batch-infrastructure/src/main/java/org/springframework/batch/item/database/builder/JdbcPagingItemReaderBuilder.java

```diff
@@ -331,7 +331,7 @@
 	private PagingQueryProvider determineQueryProvider(DataSource dataSource) {
```

Comment:
This method could probably be static...

Reply:
If I set that method to static then I'd need to pass in `selectClause`, `fromClause`, `whereClause`, `groupClause`, `sortKeys`, as they are referenced using `this`. But I guess at that point, might as well use `this.dataSource` without a parameter and keep it non static. Let me know...

Reply:
You're right.  That's what I get from reading just the diff.  Looks good to me. Thanks for the updates!